src/test/fuzz/addrman.cpp

```diff
@@ -38,6 +38,16 @@ FUZZ_TARGET_INIT(addrman, initialize_addrman)
     SetMockTime(ConsumeTime(fuzzed_data_provider));
     CAddrManDeterministic addr_man;
     addr_man.MakeDeterministic(ConsumeUInt256(fuzzed_data_provider));
+    if (fuzzed_data_provider.ConsumeBool()) {
+        const std::vector<uint8_t> serialized_data{ConsumeRandomLengthByteVector(fuzzed_data_provider)};
+        CDataStream ds(serialized_data, SER_DISK, INIT_PROTO_VERSION);
+        const auto ser_version{fuzzed_data_provider.ConsumeIntegral<int32_t>()};
+        ds.SetVersion(ser_version);
+        try {
+            ds >> addr_man;
+        } catch (const std::ios_base::failure&) {
+        }
```

Comment:
I think that `addr_man` is not in a usable state if an exception is thrown (i.e. it is in an undefined state). Below we call its methods and expect they would work. Maybe:
```suggestion
        try {
            ds >> addr_man;
        } catch (const std::ios_base::failure&) {
            addr_man.Clear();
        }
```

Reply:
Good catch. Fixed.